splunk_connect_for_snmp_poller/manager/poller_utilities.py

```diff
@@ -0,0 +1,186 @@
+#
+# Copyright 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import csv
+import logging.config
+from dataclasses import dataclass
+
+import schedule
+from pysnmp.hlapi import ObjectIdentity, ObjectType, UdpTransportTarget, getCmd
+from splunk_connect_for_snmp_poller.manager.realtime.oid_constant import (
+    OidConstant,
+)
+from splunk_connect_for_snmp_poller.manager.realtime.real_time_data import (
+    should_redo_walk,
+)
+from splunk_connect_for_snmp_poller.manager.tasks import snmp_polling
+from splunk_connect_for_snmp_poller.manager.validator.inventory_validator import (
+    is_valid_inventory_line_from_dict,
+    should_process_inventory_line,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InventoryRecord:
+    host: str
+    version: str
+    community: str
+    profile: str
+    frequency_str: str
+
+
+def _should_process_current_line(inventory_record):
+    return should_process_inventory_line(
+        inventory_record.host
+    ) and is_valid_inventory_line_from_dict(
+        inventory_record.host,
+        inventory_record.version,
+        inventory_record.community,
+        inventory_record.profile,
+        inventory_record.frequency_str,
+    )
+
+
+def onetime_task(host, version, community, profile, server_config, splunk_indexes):
+    logger.debug(
+        f"Executing onetime_task for {host} version={version} community={community} profile={profile}"
+    )
+
+    snmp_polling.delay(
+        host,
+        version,
+        community,
+        profile,
+        server_config,
+        splunk_indexes,
+        one_time_flag=True,
+    )
+    return schedule.CancelJob
+
+
+def parse_inventory_file(inventory_file_path):
+    with open(inventory_file_path, newline="") as inventory_file:
+        for agent in csv.DictReader(inventory_file, delimiter=","):
+            inventory_record = InventoryRecord(
+                agent["host"],
+                agent["version"],
+                agent["community"],
+                agent["profile"],
+                agent["freqinseconds"],
+            )
+            if _should_process_current_line(inventory_record):
+                yield inventory_record
+
+
+def _extract_sys_uptime_instance(
```

Comment:
It might be my Java habits speaking, but I think it would be a good idea to abstract all actual SNMP connection code and all scheduler related code into separate modules/classes

Reply:
Strictly speaking about SNMP, I don't know what else we could do. The PySnmp library is already an high-level API. We have a separate unit-tested module for handling hostname/post format, ContexData, and all authorisation/authentication -related stuff.